plugin/trino-mongodb/src/main/java/io/trino/plugin/mongodb/MongoSession.java

```diff
@@ -823,34 +834,62 @@ private String toRemoteSchemaName(String schemaName)
         if (!caseInsensitiveNameMatching) {
             return schemaName;
         }
-        for (String remoteSchemaName : client.listDatabaseNames()) {
+        for (String remoteSchemaName : listDatabaseNames()) {
             if (schemaName.equals(remoteSchemaName.toLowerCase(ENGLISH))) {
                 return remoteSchemaName;
             }
         }
         return schemaName;
     }
 
+    private Iterable<String> listDatabaseNames()
+    {
+        OperationExecutor operationExecutor = ((MongoClientImpl) client).getOperationExecutor();
+        CodecRegistry codecRegistry = ((MongoClientImpl) client).getSettings().getCodecRegistry();
+        boolean retryReads = ((MongoClientImpl) client).getSettings().getRetryReads();
+        return new ListDatabasesIterableImpl<>(null, BsonDocument.class, codecRegistry, ReadPreference.primary(), operationExecutor, retryReads)
+                .nameOnly(true)
+                .authorizedDatabasesOnly(authorizedDatabases)
+                .map(result -> result.getString("name").getValue());
+    }
+
     private String toRemoteTableName(String schemaName, String tableName)
     {
         verify(tableName.equals(tableName.toLowerCase(ENGLISH)), "tableName not in lower-case: %s", tableName);
         if (!caseInsensitiveNameMatching) {
             return tableName;
         }
-        for (String remoteTableName : client.getDatabase(schemaName).listCollectionNames()) {
+        for (String remoteTableName : listCollectionNames(schemaName)) {
             if (tableName.equals(remoteTableName.toLowerCase(ENGLISH))) {
                 return remoteTableName;
             }
         }
         return tableName;
     }
 
+    private Iterable<String> listCollectionNames(String databaseName)
+    {
+        Document listCollectionsCommand = new Document(ImmutableMap.<String, Object>builder()
```

Comment:
Can we make this command as a static final variable ?

Reply:
Because `Document` is mutable I extracted ImmutableMap